plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -214,10 +215,13 @@ StorageOptions createStorageOptions(
             storageOptionsBuilder.setProjectId(clientSettings.getProjectId());
         }
         if (clientSettings.getCredential() == null) {
-            logger.warn(
-                "\"Application Default Credentials\" are not supported out of the box."
-                    + " Additional file system permissions have to be granted to the plugin."
-            );
+            logger.info("\"Application Default Credentials\" will be in use");
+            try {
```

Comment:
As far as I can tell, this is not going to work, the DefaultCredentialsProvider looks into file system (CREDENTIAL_ENV_VAR, getWellKnownCredentialsFile) that won't be accessible to the GCP plugin.

Reply:
But assuming "additional file system permissions were granted" this code change does enable using default credentials, doesn't it?

Reply:
Thanks for your time to reviewing this, yes this code change does enable using default credentials. Will work on adding a test and repost.